tx_json.go

```diff
@@ -0,0 +1,415 @@
+package wavelet
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/perlin-network/wavelet/sys"
+	"github.com/pkg/errors"
+	"github.com/valyala/fastjson"
+)
+
+var (
+	// ErrNoTag defines an error describing an empty Tag.
+	ErrNoTag = errors.New("no tag specified")
+
+	// ErrInvalidTag defines an error describing an invalid TransactionTag.
+	ErrInvalidTag = errors.New("tag is invalid")
+
+	// ErrCouldNotParse defines an error describing the inability to parse a given payload.
+	ErrCouldNotParse = errors.New("could not parse the given payload")
+
+	// ErrNilField defines an error describing a field value equal to nil.
+	ErrNilField = errors.New("field is nil")
+
+	// ErrInvalidOperation defines an error describing an invalid operation value.
+	ErrInvalidOperation = errors.New("operation is invalid")
+
+	// ErrInvalidAccountIDSize defines an error describing an invalid account ID size.
+	ErrInvalidAccountIDSize = errors.New("account ID is of an invalid size")
+)
+
+/* BEGIN EXPORTED METHODS */
+
+// ParseJSON parses the given JSON payload input.
+func ParseJSON(data []byte, tag string) ([]byte, error) {
+	if tag == "" { // Check no transaction tag
+		return nil, ErrNoTag // Return no tag error
+	}
+
+	if err := fastjson.ValidateBytes(data); err != nil { // Check not valid JSON
+		return nil, err // Return found error
+	}
+
+	valid := false // Initialize valid buffer
+
+	for i := 0; i < 5; i++ { // Iterate through tags
+		if tag == strconv.Itoa(int(sys.Tag(i))) { // Check valid tag
+			valid = true // Set true
+
+			tag = sys.Tag(i).String() // Set tag to formal name
+		}
+	}
+
+	if !valid { // Check invalid
+		return nil, ErrInvalidTag // Return error
+	}
+
+	switch tag { // Handle different tag types
```

Comment:
I believe we should make `TagToLabels` and `LabelsToTags` map in the `sys` pkg so we don't have to repeat the actual label values here.

Reply:
Seems like the `sys.TagConversiono` map already addresses this issue. Could you elaborate a bit more on what you mean?

Reply:
Ah yep, in that case it's a matter of not having to repeat the label values "nop", "transfer", etc. by using the `sys.TagConversion` map or `String()` method for tags.